### PR TITLE
Add draft for Flutter Driver post

### PR DESCRIPTION
**Pull request by pascalw**
The purpose of the post is to show how Flutter Driver can be extended with custom commands. I struggled a bit with the example used in this post though, as it requires a bit of setup that is not super relevant to the topic. I wonder if it doesn't distract to much from what I try to convey?

### CODE REVIEW

File: draft/flutter-driver.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
+# Extending Flutter Driver with custom commands
+
+[Flutter Driver](https://flutter.dev/docs/cookbook/testing/integration/introduction) is a library to write end-to-end integration tests for Flutter apps. It's similar to Selenium WebDriver (for web apps), Espresso (for native Android apps) and Earl Grey (for native iOS apps). It works by instrumenting the Flutter app and then driving the application using a suite of Dart tests.
```
Comment (drumusician) [inline draft/flutter-driver.md]: maybe also link to the other e2e test tools?

File: draft/flutter-driver.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      // all other MaterialApp initialisation here
+      navigatorObservers: navigatorObserver == null ? [] : [navigatorObserver],
+    );
+  }
+}
+```
+
+What this does is allow us to hook up a NavigationObserver from our test wrapper like so:
```
Comment (drumusician) [inline draft/flutter-driver.md]:
```suggestion
This allows us to hook up a NavigationObserver from our test wrapper like so:
```

File: draft/flutter-driver.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+void main() {
+  FlutterDriver driver;
+
+  setUpAll(() async {
+    driver = await FlutterDriver.connect();
+    TestHelper.driver = driver;
+  });
+
+  tearDownAll(() async {
+    driver?.close();
```
Comment (drumusician) [inline draft/flutter-driver.md]: Maybe good to explain the ? here?

Comment (thnukid) [reply draft/flutter-driver.md]: https://www.woolha.com/tutorials/dart-using-null-aware-operators-examples

Comment (pascalw) [reply draft/flutter-driver.md]: It's kind of besides the point of the post to explain Dart language constructs. So for brevity I rather not go into this.

File: draft/flutter-driver.md (hunk 1 above)
Comment (pascalw) [reply draft/flutter-driver.md]: ✅

You can actually use any (Dart) application to interact with Flutter Driver, but it's common to use the Dart test package to write unit tests using the driver. I linked to the Dart test package now.

**Comment by pascalw** [comment]
Thanks for the feedback!

**Pull request closed without merging**